### PR TITLE
fix problems discovered during the seminar

### PR DESCRIPTION
**Pull request by h-2**
and move code snippets into files.

### CODE REVIEW

**Comment by h-2** [comment]
@smehringer  Just some very minor fixes, but they conflict with #810 so please only merge one of them and ping me again! thx

**Comment by h-2** [comment]
now rebased on recent changes

File: doc/tutorial/ranges/index.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ C++20 introduces the notion of *ranges* and provides algorithms that accept such
 `std::ranges::sort(v)` now works if `v` is range – and vectors are ranges!
 
 What about the two examples that suggest superiority of the iterator-based approach? In C++20 you can do the following:
-  * sorting only all elements after the fifth one: `std::ranges::sort(std::view::drop(v, 5))`
+  * sorting only all elements after the fifth one: `std::ranges::sort(seqan3::view::drop(v, 5))`
```
Comment (eseiler) [inline doc/tutorial/ranges/index.md]: Isn't it `std::view::drop` ?

File: doc/tutorial/ranges/range_snippets.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+#include <vector>
+#include <range/v3/view/drop.hpp>
+#include <seqan3/range/view/all.hpp>
+#include <seqan3/std/view/all.hpp>
+
```
Comment (eseiler) [inline doc/tutorial/ranges/range_snippets.cpp]: Do you need all these includes?
is `#include <seqan3/std/ranges>` enough?

File: doc/tutorial/ranges/range_solution3.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#include <iostream>
+#include <seqan3/range/view/all.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/std/ranges>
+
```
Comment (eseiler) [inline doc/tutorial/ranges/range_solution3.cpp]: Same here

File: doc/tutorial/ranges/range_snippets.cpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+int main()
+{
+{
+//![all]
+//![def]
+std::vector vec{1, 2, 3, 4, 5, 6};
+//![rev_def]
+auto v = seqan3::view::reverse(vec);
```
Comment (smehringer) [inline doc/tutorial/ranges/range_snippets.cpp]: std::reverse

File: doc/tutorial/ranges/range_snippets.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+std::cout << *v.begin() << '\n';
+//![all]
+}
+
+{
+//![assign_through]
+//![piped]
+std::vector vec{1, 2, 3, 4, 5, 6};
+auto v = vec | seqan3::view::reverse | ranges::view::drop(2);
```
Comment (smehringer) [inline doc/tutorial/ranges/range_snippets.cpp]: std::reverse, std::drop

File: doc/tutorial/ranges/index.md (hunk 1 above)
Comment (h-2) [reply doc/tutorial/ranges/index.md]: yep

File: doc/tutorial/ranges/range_snippets.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+std::cout << *v.begin() << '\n';
+//![piped]
+*v.begin() = 42;                  // now vec == {1, 2, 3, 42, 5, 6 } !!
+//![assign_through]
+}
+
+{
+//![solution1]
+std::vector vec{1, 2, 3, 4, 5, 6};
+auto v = vec | seqan3::view::filter(   [] (auto const i) { return i % 2 == 0; })
```
Comment (smehringer) [inline doc/tutorial/ranges/range_snippets.cpp]: std::filter, std::transform

File: doc/tutorial/ranges/range_snippets.cpp (hunk 2 above)
Comment (h-2) [reply doc/tutorial/ranges/range_snippets.cpp]: no, range/view/all includes our views, std/view/all includes the standard's views

Comment (h-2) [reply doc/tutorial/ranges/range_snippets.cpp]: drop is superfluous though

File: doc/tutorial/ranges/range_solution3.cpp [hunk 7]
```diff
@@ -0,0 +1,23 @@
+#include <iostream>
```
Comment (smehringer) [inline doc/tutorial/ranges/range_solution3.cpp]: do you need this?

File: doc/tutorial/ranges/range_solution3.cpp [hunk 8]
```diff
@@ -0,0 +1,22 @@
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/all.hpp>
```
Comment (smehringer) [inline doc/tutorial/ranges/range_solution3.cpp]: Do you need this?

File: doc/tutorial/ranges/range_solution3.cpp [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+    auto s_as_dna = s | seqan3::view::char_to<seqan3::dna5>;
+    // Bonus:
+    //auto s_as_dna = s | std::view::transform([] (char const c)
+    //{
+    //    return seqan3::assign_char_strict(seqan3::dna5{}, c);
+    //});
+
+    seqan3::debug_stream << "Original: " << s_as_dna << '\n';
+    seqan3::debug_stream << "RevComp:  " << (s_as_dna | std::view::reverse | seqan3::view::complement) << '\n';
```
Comment (smehringer) [inline doc/tutorial/ranges/range_solution3.cpp]: You did not include view::complement or view::translate... :thinking: probably somewhere included in the debug_stream header. For the tutorial I would include the headers explicitly

Comment (smehringer) [reply doc/tutorial/ranges/range_solution3.cpp]: EDIT: ops, you include `range/all.hpp` :woman_facepalming:  can you add a comment behind the include? I actually thought you included the view::all of ranges

**Pull request merged**